LeanKit-Labs/seriate
# Feature tidy up

Might make sense to review these changes commit by commit, so that there is some context to the changes

Removed unused dependencies
Removed dependencies with security vulnerabilities
Updated a few dependencies
Replaced monologue with EventEmitter
Removed metrics instrumentation dev dependency
Fixed some eslint things
Converted to use native promises
Consolidated to use shared auto required test setup

Testing steps:

1. Pull down the project and run the tests
2. Open a project that uses seriate, and npm link seriate to a local version of this PR.
3.  Regression test that all the functionality works in the project the way that it should, and no errors are encounter relating to seriate.
